3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/ProductModificationAPI.java

```diff
@@ -0,0 +1,46 @@
+package com._6core.platform.warehouse.rest.v1;
+
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductDescriptionRequest;
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductDescriptionResponse;
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductNameResponse;
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductNameRequest;
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductPriceRequest;
+import com._6core.platform.warehouse.rest.v1.dto.UpdateProductPriceResponse;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import reactor.core.publisher.Mono;
+
+/**
+ * Interface for modifying product information in the Warehouse Service.
+ */
+
+@Tag(name = "Product Modification", description = "REST specification for modifying product information")
+public interface ProductModificationAPI<T, K> {
```

Comment:
Using generics like here is a bad idea. 
1. We don't have a clear understanding of ApiResponse(content=@Content(...)) for 200 code
2. summary/description will never cover all generic cases (of request or response) or will be ugly. 
3. You will get a spring error that URI must be different for different endpoints
```suggestion
public interface ProductModificationApi {
```

Reply:
Ok, thank you.